internal/app/machined/internal/server/v1alpha1/v1alpha1_server.go

```diff
@@ -181,6 +181,36 @@ func (s *Server) Reset(ctx context.Context, in *machine.ResetRequest) (reply *ma
 	return reply, nil
 }
 
+// Recover recovers the control plane.
+//
+// nolint: dupl
+func (s *Server) Recover(ctx context.Context, in *machine.RecoverRequest) (reply *machine.RecoverResponse, err error) {
+	log.Printf("recover request received")
+
+	if s.Controller.Runtime().Config().Machine().Type() == runtime.MachineTypeJoin {
+		return nil, fmt.Errorf("recover can only be performed on a control plane node")
```

Comment:
I can't get this to actually return back to the client for some reason. This follows the same pattern as other RPCs so I don't really get it.